src/library_gl.js

```diff
@@ -728,7 +728,11 @@ var LibraryGL = {
         ret = allocate(intArrayFromString(gl_exts.join(' ')), 'i8', ALLOC_NORMAL);
         break;
       case 0x8B8C /* GL_SHADING_LANGUAGE_VERSION */:
+#if USE_WEBGL2
+        ret = allocate(intArrayFromString('OpenGL ES GLSL 3.00 (WebGL2)'), 'i8', ALLOC_NORMAL);
```

Comment:
Hmm, this is not good, since `-s USE_WEBGL2` does not mean that we will necessarily be using WebGL 2, but it means that a) we generate a build that can create a WebGL2 context and b) by default we will create a WebGL 2 context from APIs that don't otherwise express a choice. Still it is possible to create and use both WebGL 1 or WebGL 2 at runtime when `-s USE_WEBGL2` is passed, so we should detect here if it's actually a WebGL 2 context that we got, and only then return "OpenGL ES GLSL 3.00 (WebGL2)" if so.

Although I wonder why we have even hardcoded this in the past. I think what we should be doing is this:

	case 0x8B8C /* GL_SHADING_LANGUAGE_VERSION */:
		var glslVersion = GLctx.getParameter(GLctx.SHADING_LANGUAGE_VERSION);
		// Map WebGL GL_SHADING_LANGUAGE_VERSION string format to GLES format.
		if (glslVersion.indexOf('WebGL GLSL ES 1.0') != -1) glslVersion = 'OpenGL ES GLSL 1.00 (WebGL)';
	#if USE_WEBGL2
		else if (glslVersion.indexOf('WebGL GLSL ES 3.00') != -1) glslVersion = 'OpenGL ES GLSL 3.00 (WebGL 2)';
	#endif
		ret = allocate(intArrayFromString(glslVersion), 'i8', ALLOC_NORMAL);
		break;

so that if an implementation returns something really odd past the spec, it will be caught. What do you think?